cnf-certification-test/lifecycle/podrecreation/podrecreation.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (C) 2020-2022 Red Hat, Inc.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package podrecreation
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+	"github.com/test-network-function/cnf-certification-test/internal/clientsholder"
+	"github.com/test-network-function/cnf-certification-test/pkg/provider"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ReplicaSetString  = "ReplicaSet"
+	DeploymentString  = "Deployment"
+	StatefulsetString = "StatefulSet"
+	DaemonSetString   = "DaemonSet"
+)
+
+func CordonNode(name string) error {
```

Comment:
Could we create a `podrecreation` struct here to store stuff like the client, etc?  Then funcs like `CordonNode`, `Uncordon`, etc. could be implemented as part of the struct.  Corresponding interfaces could be created for it too.